### PR TITLE
Layout Mii parameters input/output, and return success as result of applet work

### PR DESCRIPTION
**Pull request by mailwl**
Main reason of PR - make Crosswords Plus working

### CODE REVIEW

**Comment by wwylele** [comment]
Nice, Mario 3D Land now can bypass the mii selecting and go ingame without a save file!

File: src/core/hle/applets/mii_selector.cpp [hunk 1]
```diff
@@ -55,6 +55,9 @@ ResultCode MiiSelector::StartImpl(const Service::APT::AppletStartupParameter& pa
     // TODO(Subv): Set the expected fields in the response buffer before resending it to the application.
     // TODO(Subv): Reverse the parameter format for the Mii Selector
 
+    MiiResult* mii_result = reinterpret_cast<MiiResult*>(parameter.data);
```
Comment (lioncash) [inline src/core/hle/applets/mii_selector.cpp]: This cast is considered undefined behavior. Please use memcpy or something that doesn't require punning memory.

Comment (mailwl) [inline src/core/hle/applets/mii_selector.cpp]: i don't know... in this case it really safe, maybe add assert check to struct size?
memcpy/memset not so obvious as named assigning

Comment (lioncash) [inline src/core/hle/applets/mii_selector.cpp]: No. It *is* undefined regardless of what you think it is in this context. This is within the standard. It breaks strict aliasing. Please use the correct means of copying data (i.e. memcpy).

**Comment by mailwl** [comment]
@lioncash done, but now both aligned structs not used, may i keep it? at least for memo (types and sizes reversed and need further investigation)

**Comment by lioncash** [comment]
@mailwl Yes, that's fine.

File: src/core/hle/applets/mii_selector.h [hunk 2]
```diff
@@ -16,6 +16,50 @@
...
 
+struct MiiConfig {
+    u8  unk_000;
+    u8  unk_001;
+    u8  unk_002;
+    u8  unk_003;
+    u8  unk_004;
+    INSERT_PADDING_BYTES(3);
+    u16 unk_008;
+    INSERT_PADDING_BYTES(0x8c - 0xA);
```
Comment (Subv) [inline src/core/hle/applets/mii_selector.h]: 0x8C

File: src/core/hle/applets/mii_selector.h [hunk 3]
```diff
@@ -16,6 +16,50 @@
...
+ASSERT_REG_POSITION(unk_090, 0x90);
+ASSERT_REG_POSITION(unk_094, 0x94);
+ASSERT_REG_POSITION(unk_0FE, 0xFE);
+#undef ASSERT_REG_POSITION
+
+struct MiiResult {
+    u32 result_code;
+    u8  unk_04;
+    INSERT_PADDING_BYTES(7);
+    u8 unk_0c[0x60];
```
Comment (Subv) [inline src/core/hle/applets/mii_selector.h]: 0C

File: src/core/hle/applets/mii_selector.h [hunk 4]
```diff
@@ -16,6 +16,50 @@
...
+    u8 unk_6c[0x16];
```
Comment (Subv) [inline src/core/hle/applets/mii_selector.h]: 6C

**Comment by Subv** [comment]
LGTM other than the few nitpicks. I'd like to get this one merged as soon as possible

**Comment by suelo50** [comment]
pokemon rumble world work? because this game requires player mii  to start

**Comment by mailwl** [comment]
done.
@suelo50, IDK, eur version(CTR-P-ECFA) doesn't try to use mii selector, and stuck at window with broken font:
prw

**Comment by Subv** [comment]
LGTM

**Comment by suelo50** [comment]
pokemon rumble world how fix it
pokemon rumble

**Comment by suelo50** [comment]
pokemon rumble world how fix it?
pokemon rumble

**Comment by suelo50** [comment]
pokemon rumble world 
pokemon rumble
fixed it?

**Pull request merged**